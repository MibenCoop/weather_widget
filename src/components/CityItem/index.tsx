import React from "react";

import { City } from "../../interfaces";

export interface CityProps {
  city: City;
}

const CityItem: any = ({ city }: CityProps) => {
  // console.log("city", city);
  return <li style={{ textAlign: "left" }}>{city.city}</li>;
};

export default CityItem;
