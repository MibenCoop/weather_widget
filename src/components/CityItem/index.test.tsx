import React from 'react';
import { shallow } from 'enzyme';

import CityItem from './index';

describe('City item component', () => {
    const cityMock = {
        city: 'Moscow',
        admin: 'Moskva',
        country: 'Russia',
        population_proper: '10452000',
        iso2: 'RU',
        capital: 'primary',
        lat: '55.752222',
        lng: '37.615556',
        population: '10452000',
    };
    it('Renders a City Item ', () => {
        const city = shallow(<CityItem city={cityMock} />);
        expect(city).toMatchSnapshot();
    });
    it('Renders a correct city name ', () => {
        const city = shallow(<CityItem city={cityMock} />);
        expect(city.find('li').text()).toEqual('Moscow');
    });
});
