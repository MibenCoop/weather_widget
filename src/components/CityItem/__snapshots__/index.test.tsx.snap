// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City item component Renders a City Item  1`] = `
<li
  style={
    Object {
      "textAlign": "left",
    }
  }
>
  Moscow
</li>
`;
